lib/puppet/util/network_device/cisco_ios/device.rb

```diff
@@ -36,18 +36,20 @@ def send_command(connection_to_use, options, debug = false)
       end
       return_value = connection_to_use.cmd(options)
       # Check for authentication related errors
-      access_denied = commands['default']['access_denied']
-      command_authorization_failed = commands['default']['command_authorization_failed']
-      error_in_authentication = commands['default']['error_in_authentication']
+      access_denied = Regexp.new(%r{#{commands['default']['access_denied']}})
```

Comment:
I believe these Regexp.new changes are redundant as %r{} is generating a regex pattern already

```ruby
irb(main):018:0> %r{^.*Command is only allowed on VLAN.*$} == Regexp.new(%r{^.*Command is only allowed on VLAN.*$})
=> true
irb(main):031:0> %r{^.*Command is only allowed on VLAN.*$}.class
=> Regexp
irb(main):032:0> Regexp.new(%r{^.*Command is only allowed on VLAN.*$}).class
=> Regexp
```